app/models/output_buffer.rb

```diff
@@ -27,6 +27,14 @@ def initialize
     @closed = false
   end
 
+  def puts(line)
```

Comment:
this seems a bit counter-intuitive ... puts xxx.strip for usages with trailing whitespace instead ?